bun/src/query/queryUtils.ts

```diff
@@ -3,66 +3,70 @@ import { ValueOf } from "type-fest";
 import type { TraverserMap, GroovyTraversal } from "groovy/dsl";
 
 /**
- * Generic return type for a gremlin traversal
- * currently supports {@link GroovyTraversal} for chaining and {@link TraverserMap} for results
+ * Generic return type for a {@link GroovyTraversal} invocation
+ * currently supports {@link GroovyTraversal} for chaining and {@link TraverserMap} for traversal results
  */
 export interface NextT {
   GT: GroovyTraversal;
   TM: TraverserMap;
 }
+
 /**
  * union type of {@link NextT} object values
  */
 export type NextTUnion = ValueOf<NextT>;
 
 /**
  * base opts for a gremlin traversal
+ * @prop end if truthy returns a traversal value, else returns a traversal for chaining
  * @prop limitX e.g. traversal.range(limitX, limitY)
  * @prop limitY e.g. traversal.range(limitX, limitY)
- * @prop end if truthy returns a traversal value, else returns a traversal for chaining
  */
 export type BaseOpts<T = Record<string, any>> = T & {
```

Comment:
we need an outer limit, and an inner limit
e.g. `getThisThing.range(0, 10).project(a,b).by(thisOtherThing.range(0, 5)`